### PR TITLE
Replace get metadata

### PR DESCRIPTION
**Pull request by flopez-bao**
**Changes**

- Referencing [DP-770](https://jira.pepfar.net/browse/DP-770), transitioned `datapackcommons::getMetaData` to `datimtuils::getMetaData`.
- Changed use in CreateDataFiles.R, left config_update_analysis.R with native use since its an incomplete script and seems to be an analysis-- will follow up with Sam when he's back on use case.
- altered `test-metadata-load.R` to account for `datimutils` use.
- removed getMetaData from dhis2Api.R, redocumented.

**Testing**
- checked results (dim_item_sets, data_required, Map22To23) of running createDataFiles.R against master branch

### CODE REVIEW

File: data-raw/CreateDataFiles.R [hunk 1]
```diff
@@ -25,8 +24,7 @@ ValidateDimItems <-
       na.omit() %>%
       paste0(collapse = ",") %>%
       {paste0("id:in:[", ., "]")} %>% # nolint: brace_linter.
-      {datapackcommons::getMetadata("dimensions", filters = ., fields = "items[name,id]")} %>% # nolint: brace_linter.
-      tidyr::unnest(c("items"))
```
Comment (JordanBalesBAO) [inline data-raw/CreateDataFiles.R:29]: Do we not need to unnest here?

File: data-raw/config_update_analysis.R [hunk 2]
```diff
@@ -57,7 +60,8 @@ parseIndicators <- function(indicator_uids) {
 
 doMC::registerDoMC(cores = 5)
 
-datimutils::loginToDATIM("~/.secrets/datim.json")
+datimutils::loginToDATIM(paste0(Sys.getenv("SECRETS_FOLDER"),
```
Comment (JordanBalesBAO) [inline data-raw/config_update_analysis.R:63]: brilliant

**Review by JordanBalesBAO — COMMENTED** [review]
Looks good, just need to confirm my comment and I can approve.  There also seems to be a git permissions issue that is causing the ci to fail. We might want to mention this in standup

File: data-raw/CreateDataFiles.R (hunk 1 above)
Comment (flopez-bao) [reply data-raw/CreateDataFiles.R:29]: So it seems with `datimutils::getMetaData` the need to unest is unecesarry. If you could quickly test to confirm would be good. Basically I compared the `api_names_ids` result with datimutils function  to the native one.

**Review by JordanBalesBAO — APPROVED** [review]
passed all tests I threw at it.

File: data-raw/CreateDataFiles.R (hunk 1 above)
Comment (cnemarich) [reply data-raw/CreateDataFiles.R:29]: Was this confirmed?

**Comment by cnemarich** [comment]
> Looks good, just need to confirm my comment and I can approve. There also seems to be a git permissions issue that is causing the ci to fail. We might want to mention this in standup

@JordanBalesBAO I've created a ticket for this that is with the Systems Engineering team.

**Review by cnemarich — APPROVED** [review]

File: tests/testthat/test-metadata-load.R [hunk 3]
```diff
@@ -6,15 +6,16 @@ test_that("We can load data elements", {
                    handle = httr::handle("https://play.dhis2.org/2.30/"))
   test_filters <- "name:like:HIV positive"
   test_fields <- c("id", "name", "shortName")
-  test_de_payload <- getMetadata("dataElements",
+  test_de_payload <- datimutils::getMetadata("dataElements",
```
Comment (cnemarich) [inline tests/testthat/test-metadata-load.R:9]: Do we even need a test here if this function is being tested upstream in the `datimutils` package itself?

**Review by cnemarich — COMMENTED** [review]
See my notes on `getMetaData` test

File: tests/testthat/test-metadata-load.R (hunk 3 above)
Comment (flopez-bao) [reply tests/testthat/test-metadata-load.R:9]: I was wondering the same thing. I will remove it @cnemarich.

**Pull request merged**